internal/weldr/api.go

```diff
@@ -401,7 +409,7 @@ func (api *API) checkImageTypeDenylist(imageType string) error {
 // This is necessary because different distros support different image types, and the image
 // type may have a different package set than other distros.
 func (api *API) getImageType(distroName, imageType string) (distro.ImageType, error) {
-	if err := api.checkImageTypeDenylist(imageType); err != nil {
+	if err := api.checkImageTypeDenylist(distroName, imageType); err != nil {
 		return nil, err
```

Comment:
If you choose to change the return type to `bool` still return an error here, just move the one from inside the current function.

Then in `composeHandler` I think it needs to include the error string from `getImageType` because we now have 3 ways to exit this and all of them show up as the same `Unknown compose type for architecture: %s` response which is going to start getting confusing (likely my fault for not doing that when I added a new error result).

Reply:
I agree regarding `composeHandler`. I decided to keep the API (response) changes minimal, by not propagating the error message from `getImageType()` to the API response, because there were already two distinct errors which were not propagated. But it definitely make sense to distinguish these situations by returning the appropriate error message.

Reply:
This is also done.